sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -585,25 +585,26 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
 
   if (NumDevices)
     *NumDevices = ZeDeviceCount;
-
-  try {
-    // TODO: Delete array at teardown
-    ze_device_handle_t *ZeDevices = new ze_device_handle_t[ZeDeviceCount];
-    ZE_CALL(zeDeviceGet(ZeDriver, &ZeDeviceCount, ZeDevices));
-
-    for (uint32_t I = 0; I < ZeDeviceCount; ++I) {
-      if (I < NumEntries) {
-        Devices[I] = new _pi_device(ZeDevices[I], Platform);
-        pi_result Result = Devices[I]->initialize();
-        if (Result != PI_SUCCESS) {
-          return Result;
+  if (NumEntries > 0) {
```

Comment:
Please check "Devices" instead (and assert that if "Devices" is not NULL, then NumEntries  > 0)
NOTE: semantics is from here: https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clGetDeviceIDs.html

Reply:
Also, let's just return here if no further processing is needed (to avoid large indented block)